shotgun_api3/shotgun.py

```diff
@@ -2365,6 +2365,20 @@ def _upload_to_sg(self, entity_type, entity_id, path, field_name, display_name,
 
         params.update(self._auth_params())
 
+        # If we ended up with a unicode string path, we need to encode it
+        # as a utf-8 string. If we don't, there's a chance that there will
+        # will be an attempt later on to encode it as an ascii string, and
+        # that will fail ungracefully if the path contains any non-ascii
+        # characters.
+        if isinstance(path, unicode):
+            try:
+                path = path.encode("utf-8")
+            except UnicodeEncodeError:
```

Comment:
I might be wrong, but I don't think we need this exception check. All unicode characters can be turned into utf-8 (_UTF-8 is a variable width character encoding capable of encoding all 1,112,064 valid code points in Unicode using one to four 8-bit bytes._ - see https://en.wikipedia.org/wiki/UTF-8)

Reply:
Cool. I was doing it out of an over abundance of caution, but if it's completely unnecessary then I'll remove it!